### PR TITLE
Fun/Consumed Budget Ratio metric

### PR DESCRIPTION
**Pull request by mrtergl**
Added metric would be useful in monitoring tools such as Prometheus/Grafana. It helps us better understand how much of the budget we are spending

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -121,7 +121,7 @@ dev =
     pre-commit
     bandit
     GitPython >=3.1.35  # avoid CVE-2023-41040, CVE-2023-40267 and CVE-2023-40590 (reported by `safety check`)
-    safety
+    safety >=2.3.5
```
Comment (lvaylet) [inline setup.cfg:124]: Is this necessary, as it does not seem to solve the issue of Python 3.11 tests not completing successfully?

File: slo_generator/exporters/bigquery.py [hunk 2]
```diff
@@ -276,6 +276,11 @@ def _format(errors):
         "type": "FLOAT",
         "mode": "NULLABLE",
     },
+    {
+        "name": "consumed_budget_ratio",
```
Comment (lvaylet) [inline slo_generator/exporters/bigquery.py:280]: `error_budget_consumed_ratio` to remain consistent with the naming of similar variables?

File: slo_generator/report.py [hunk 3]
```diff
@@ -60,6 +61,7 @@ class SLOReport:
     error_budget_minutes: float
     error_budget_remaining_minutes: float
     error_minutes: float
+    consumed_budget_ratio: float
```
Comment (lvaylet) [inline slo_generator/report.py:64]: same here: `error_budget_consumed_ratio`?

File: slo_generator/report.py [hunk 4]
```diff
@@ -178,6 +181,7 @@ def build(self, step, data):
             error_budget_remaining_minutes=eb_remaining_minutes,
             error_budget_minutes=eb_target_minutes,
             error_minutes=eb_minutes,
+            consumed_budget_ratio=eb_ratio,
```
Comment (lvaylet) [inline slo_generator/report.py:184]: same here: `error_budget_consumed_ratio`?

File: tests/unit/fixtures/slo_report_v2.json [hunk 5]
```diff
@@ -16,6 +16,7 @@
   "error_budget_remaining_minutes": -288,
   "error_budget_target": 0.09999999999999998,
   "error_minutes": "360.0",
+  "consumed_budget_ratio": 19.9999999999999,
```
Comment (lvaylet) [inline tests/unit/fixtures/slo_report_v2.json:19]: same here: `error_budget_consumed_ratio`?

**Review by lvaylet — CHANGES_REQUESTED** [review]
Thanks a lot for this PR. I added a few comments.

**Comment by lvaylet** [comment]
Thanks a lot @mrtergl. The code looks good. Let me just think about the consequences for current users of the BigQuery exporter. I think everything will be fine, as BigQuery allows adding more columns, but I want to run some tests first. Also, I do not want this new column to induce extra cost for users who rely a lot on the SLO Generator, with millions of calls every month.

**Comment by mrtergl** [comment]
I appreciate for your time and valuable review. @lvaylet

**Review by lvaylet — APPROVED** [review]

**Comment by lvaylet** [comment]
I will move forward with merging this to the `main` branch. Then I will use the Cloud Run Service deployed by the CI/CD pipeline to run a few tests.

**Pull request merged**